pkg/operator/controller/dns/controller.go

```diff
@@ -573,14 +576,23 @@ func (r *reconciler) createDNSProvider(dnsConfig *configv1.DNS, platformStatus *
 			log.Info("using fake DNS provider because cluster's ControlPlaneTopology is External")
 			return &dns.FakeProvider{}, nil
 		}
+
 		var err error
-		cisInstanceCRN := platformStatus.IBMCloud.CISInstanceCRN
-		if cisInstanceCRN == "" {
-			return nil, fmt.Errorf("missing cis instance crn")
-		}
-		dnsProvider, err = getIbmDNSProvider(dnsConfig, creds, cisInstanceCRN, userAgent)
-		if err != nil {
-			return nil, fmt.Errorf("failed to create IBM DNS manager: %v", err)
+		if platformStatus.IBMCloud.CISInstanceCRN != "" {
+			dnsProvider, err = getIbmDNSProvider(dnsConfig, creds, platformStatus.IBMCloud.CISInstanceCRN, userAgent, true)
+			if err != nil {
+				return nil, fmt.Errorf("failed to create ibm cis dns provider: %v", err)
+			}
+		} else if platformStatus.IBMCloud.DNSInstanceCRN != "" {
+			rExp := regexp.MustCompile(ibm.IBMResourceCRNRegexp)
```

Comment:
Better to define a package variable for the compiled regexp to avoid recompiling it every time `createDNSProvider` is called.  You can define it at the top of the file, where `log` is currently defined:
```go
var (
	log = logf.Logger.WithName(controllerName)

	// ibmResourceCRNCompiledRegexp is a regular expression of IBM resource CRNs.
	ibmResourceCRNCompiledRegexp = regexp.MustCompile(ibm.IBMResourceCRNRegexp)
)
```
...and then...
```suggestion
			rExp := regexp.MustCompile(ibmResourceCRNCompiledRegexp)
```
Alternatively, define `ibm.IBMResourceCRNRegexp` as a compiled regexp (see my previous comment).

Reply:
Defined it as a compiled regexp var.